Dockerfile

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 the Kubeapps contributors.
+# SPDX-License-Identifier: Apache-2.0
+
+FROM node:bullseye
+
+ARG KIND_VERSION="v0.14.0"
+ARG YQ_VERSION="v4.27.2"
+ARG TANZU_FW_VERSION="v0.25.0"
+
+# Dependencies
+RUN apt-get -y update && apt-get install --yes wget curl jq perl apt-transport-https ca-certificates gnupg2 software-properties-common && \
+        update-ca-certificates && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /usr/src
+
+# Install Bitnami's Readme generator
+RUN git clone --single-branch --no-tags https://github.com/bitnami-labs/readme-generator-for-helm readmenator
+RUN cd ./readmenator && npm install
+RUN npm install --global --only=production ./readmenator
+
+# Install Carvel toolset
+RUN bash -c "set -eo pipefail; wget -O- https://carvel.dev/install.sh | bash"
+
+# Install yq
+RUN wget "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64.tar.gz" -O - | tar xz && mv yq_linux_amd64 /usr/bin/yq
+
+# Install Helm
+RUN curl https://baltocdn.com/helm/signing.asc | gpg --dearmor | tee /usr/share/keyrings/helm.gpg > /dev/null
+RUN echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/helm.gpg] https://baltocdn.com/helm/stable/debian/ all main" | \
+        tee /etc/apt/sources.list.d/helm-stable-debian.list
+RUN apt-get update && apt-get install --yes helm
+
+# Install Docker
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+RUN add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/debian bullseye stable"
+RUN apt-get update && apt-get install --yes docker-ce docker-ce-cli containerd.io
```

Comment:
Readability was in mind when building this. It might take a little bit of time to do multiple `apt-get update`s, but I think the improvement it could bring is not worth the readability lost for our future selves.